pyprep/ransac.py

```diff
@@ -24,57 +23,65 @@ def find_bad_by_ransac(
-    fraction_good : float
-        Fraction of channels used for robust reconstruction of the signal.
-        This needs to be in the range [0, 1], where obviously neither 0
-        nor 1 would make sense.
-    corr_thresh : float
-        The minimum correlation threshold that should be attained within a
-        data window.
-    fraction_bad : float
-        If this percentage of all data windows in which the correlation
-        threshold was not surpassed is exceeded, classify a
-        channel as `bad_by_ransac`.
-    corr_window_secs : float
-        Size of the correlation window in seconds.
-    channel_wise : bool
-        If True the ransac will be done 1 channel at a time, if false
-        it will be done as fast as possible (more channels at a time).
+        Labels of channels to exclude as signal predictors during RANSAC
+        (i.e. channels already flagged as bad by metrics other than HF noise).
+    n_samples : int, optional
+        Number of random channel samples to use for RANSAC. Defaults to 50.
+    sample_prop : float, optional
+        Proportion of total channels to use for signal prediction per RANSAC
+        sample. This needs to be in the range [0, 1], where obviously neither 0
+        nor 1 would make sense. Defaults to 0.25 (e.g. 16 channels per sample
+        for a 64-channel dataset).
+    corr_thresh : float, optional
+        The minimum predicted vs. actual signal correlation for a channel to
+        be considered good within a given RANSAC window. Defaults to 0.75.
+    fraction_bad : float, optional
+        The minimum fraction of bad (i.e. below-threshold) RANSAC windows for a
+        channel to be considered bad-by-RANSAC. Defaults to 0.4.
+    corr_window_secs : float, optional
+        The duration (in seconds) of each RANSAC correlation window. Defaults to
+        5 seconds.
+    channel_wise : bool, optional
+        Whether RANSAC should be performed one channel at a time (lower RAM
+        demands) or in chunks of as many channels as can fit into the currently
+        available RAM (faster). Defaults to False (i.e. using the faster method).
+    random_state : int | None | np.random.RandomState, optional
+        The random seed with which to generate random samples of channels during
+        RANSAC. If random_state is an int, it will be used as a seed for RandomState.
+        If None, the seed will be obtained from the operating system
+        (see RandomState for details). Defaults to None.
 
     Returns
     -------
     bad_by_ransac : list
-        List of channels labels marked bad by ransac.
+        List containing the labels of all channels flagged as bad by RANSAC.
     channel_correlations : np.ndarray
-        Array of shape (windows,channels) holding the correlations of
-        the channels to their predicted ransac value in each of the windows.
+        Array of shape ``[windows, channels]`` containing the correlations of
```

Comment:
shapes in numpy are denoted using `()` brackets

Here you render that as a code-block. I think it'd be good to render all "pythonic" aspects of the docstring as code-blocks. such as 

```RST
``False``
``None``
``50`` (if it is an int default, or some such)
```

Whereas non-pythonic things should NOT be rendered as codeblock, such as:

```RST
 This needs to be in the range [0, 1]
```

(there, [0, 1] is not a Python list, but the mathematical notation for an inclusive interval, see: https://en.wikipedia.org /wiki/Interval_(mathematics)#Including_or_excluding_endpoints ... writing this, we should also exchange the word "range" there with "interval" ... because, see below:)
```
In [1]: 1 in range(0,1)
Out[1]: False
```

Reply:
Fixed/added to the RANSAC docstrings!